.github/workflows/style_docstr.yml

```diff
@@ -17,6 +17,7 @@ jobs:
         uses: actions/setup-python@v4
         with:
           python-version: 3.9
+          cache: 'pip'
 
       - name: Install pre-commit
```

Comment:
Do we need to use `cache-dependency-path`?  My understanding is that just `pip` will only use `requirements.txt`, but I'm not sure.  This comment applies to all the changes like this.

Reply:
Good point.  I've added most in c33039905714aa1917d88431c36e1de855c57920, but the documentation style doesn't really need it (nor do we have a requirements file for pre-commit).